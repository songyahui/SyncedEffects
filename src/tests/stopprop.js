"use hopscript"

var hh = require("hiphop")

var prg =
    <hh.module O>
      <hh.loop>
	<hh.emit O/>
	<hh.pause/>
      </hh.loop>
    </hh.module>;

var m = new hh.ReactiveMachine(prg, "foo");

m.addEventListener("O", function(evt) {
   console.log("first", evt.type);
});

m.addEventListener("O", function(evt) {
   evt.stopPropagation();
   console.log("second", evt.type);
});

m.addEventListener("O", function(evt) {
   console.log("third", evt.type);
});

exports.prg = m;
