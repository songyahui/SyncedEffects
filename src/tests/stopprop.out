foo> ;
first O
second O
--- Output: O
foo> ;
first O
second O
--- Output: O
foo> ;
first O
second O
--- Output: O
